bql/planner/tracer/trace.go

```diff
@@ -25,19 +25,32 @@ type Arguments struct {
 	Msgs []string
 }
 
+// event encapsulates a single tracing event.
 type event struct {
 	w          io.Writer
 	t          time.Time
 	tracerArgs func() *Arguments
 }
 
-var c chan *event
+// MessageTracer encapsulates the intrinsic verbosity of a given tracing message.
+type MessageTracer struct {
+	verbosity int
+}
+
+// tracerVerbosity represents the global verbosity level of the current tracer. Level 1 means minimum verbosity (printing
+// only what is crucial) while level 3 means maximum verbosity (printing all available tracing messages).
+var tracerVerbosity int
```

Comment:
did you consider using iota?

Iota is the "golang enum". Iota represents successive untypes integer constants which seems like what you need here PTAL: https://golang.org/ref/spec#Iota